pages/03-core/standalone-legend.page.tsx

```diff
@@ -0,0 +1,326 @@
+        const visible = itemIds.includes(i.id);
+        return {
+          ...i,
+          visible,
+          highlighted: visible,
+          marker: <ChartSeriesMarker color={colors[idx % colors.length]} type={"square"} visible={visible} />,
+        };
+      });
+    });
+  };
+
+  return (
+    <Page
+      title="Standalone legend demo"
+      subtitle="The page demonstrates the use of the standalone legend."
+      settings={<PageSettingsForm selectedSettings={["showLegendTitle", "showLegendActions"]} />}
+    >
+      <SpaceBetween direction="vertical" size="m">
+        <StandaloneLegend
+          items={items}
+          {...legendProps}
+          onClearHighlight={() => {
+            pieChartAPI.current?.clearChartHighlight();
+            lineChartAPI.current?.clearChartHighlight();
+            clearLegendHighlight();
+          }}
+          onItemHighlight={(e) => {
+            pieChartAPI.current?.highlightItems([e.detail.item.id]);
+            lineChartAPI.current?.highlightItems([e.detail.item.id]);
+            highlightLegendItem(e.detail.item.id);
+          }}
+          onVisibleItemsChange={(e) => {
+            pieChartAPI.current?.setItemsVisible(e.detail.items);
+            lineChartAPI.current?.setItemsVisible(e.detail.items);
+            changeVisibleLegentItems(e.detail.items);
+          }}
+          getLegendTooltipContent={({ legendItem }) => ({
+            header: (
+              <div>
+                <div style={{ display: "flex" }}>
+                  {legendItem.marker}
+                  {legendItem.name}
+                </div>
+              </div>
+            ),
+            body: (
+              <>
+                <table>
+                  <tbody style={{ textAlign: "left" }}>
+                    <tr>
+                      <th scope="row">Period</th>
+                      <td>15 min</td>
+                    </tr>
+                    <tr>
+                      <th scope="row">Statistic</th>
+                      <td>Average</td>
+                    </tr>
+                    <tr>
+                      <th scope="row">Unit</th>
+                      <td>Count</td>
+                    </tr>
+                  </tbody>
+                </table>
+              </>
+            ),
+            footer: (
+              <Link external={true} href="https://example.com/" variant="primary">
+                Learn more
+              </Link>
+            ),
+          })}
+        />
+        <CoreChart
+          {...omit(chartProps.pie, "ref")}
+          highcharts={Highcharts}
+          chartHeight={400}
+          legend={{ enabled: false }}
+          callback={(chart) => {
+            pieChartAPI.current = chart;
+          }}
+          onHighlight={(e) => {
+            if (e.detail.isApiCall) {
+              return;
+            }
+            if (e.detail.point) {
+              lineChartAPI.current?.highlightItems([e.detail.point.name]);
```

Comment:
nit: This particular integration can be hard to scale. That is not a problem in this particular demo of course, but it would be nice to refactor it so that it can theoretically include more charts than just two. In that case the demo page can be used as a guidance for implementing similar experiences.

One idea I have is implementing helper functions like

```
function highlightItems(sourceName, itemName) {
  for (const source of sources) {
    if (source.name !== sourceName) {
      source.api.highlightItems([itemName])
    }
  }
}
```

In this example, all charts are represented with an array of sources, which can be e.g. of type `{ name: string, api: ChartAPI }[]`.

Reply:
The helpers can also take care of isApiCall constraint etc.